desktop/src/main/java/bisq/desktop/util/validation/PhoneNumberValidator.java

```diff
@@ -49,13 +54,18 @@ public class PhoneNumberValidator extends InputValidator {
     // but isoCountryCode must be set before validation.
     public PhoneNumberValidator() {
     }
-    
+
     public PhoneNumberValidator(String isoCountryCode) {
         this.isoCountryCode = isoCountryCode;
         this.callingCode = CountryCallingCodes.getCallingCode(isoCountryCode);
         this.normalizedCallingCode = CountryCallingCodes.getNormalizedCallingCode(isoCountryCode);
     }
 
+    public PhoneNumberValidator(String isoCountryCode, int requiredLength) {
+        this(isoCountryCode);
```

Comment:
Another approach to calling this alternative constructor with a 'requiredLength' argument is to modify the CountryCallingCodes.java Map itself to incorporate requiredLength into each country's definition there along with the country code. Or make a separate Map file.  As here, if the requiredLength is not known (or missing if we setup an alternative Map), setting it to 0 will skip the requiredLength validation. If you want to go with that approach instead, I'm only setting the required length for US, Canada, and Russia that I'm sure about...

Reply:
Is this method used already in the app? I have not seen it in the PR. Or did you plan to apply it in another PR?

I would consider a map instead of a custom constructor better. I think a single map of country code to requiredLength should be enough, and if no entry present we ignore the check.

Reply:
I have a pull request coming soon for you to add a new Bisq payment method that will use this feature, but as these are actually independent things, this pull request is a **_prerequisite_** as I need to know how you want this implemented in the app before I submit an actual usage of it. 

As for **displayString**, every other string among the **2,669** strings that (already) exist is in **_every_** language file, so it would be inconsistent to not add to all language files here, so I'm going to ask you to reconsider that decision.

However, accordingly, I will revise this pull request to add a new source module with a **country-code -> requiredLength** map as you suggest, as well as use the 'optional' flag since not all countries may have a consistent phone number length (or we don't know what it is).

Reply:
"As for displayString, every other string among the 2,669 strings that (already) exist is in every language file, so it would be inconsistent to not add to all language files here." - the translations would be handled via transifex. Though I am not sure how well it is maintained these days. If you like you can add all translations (using some AI translation tool). But to just add the english is redundant and give the wrong impression as it was translated (I think in transifex they see missing entries, but not if its in english).

Reply:
> the translations would be handled via transifex. Though I am not sure how well it is maintained these days. 
> If you like you can add all translations (using some AI translation tool). 
> But to just add the english is redundant and give the wrong impression as it was translated (I think in transifex they see missing entries, but not if its in english).

This is off-topic, but a rough analysis of the language files which use foreign characters reveals the current level of translation in them is pretty low and there is already a ton of these redundant English strings in them:
```
displayStrings_cs.properties: translatedcount=1542, percent=57.77%
displayStrings_fa.properties: translatedcount=1556, percent=58.30%
displayStrings_ja.properties: translatedcount=2050, percent=76.81%
displayStrings_pl.properties: translatedcount=1364, percent=51.11%
displayStrings_ru.properties: translatedcount=1585, percent=59.23%
displayStrings_th.properties: translatedcount=1438, percent=53.88%
displayStrings_vi.properties: translatedcount=1445, percent=54.14%
displayStrings_zh-hans.properties: translatedcount=2016, percent=75.53%
displayStrings_zh-hant.properties: translatedcount=1991, percent=74.60%

totalcount=2669, overall average 62.39% translated
```
The above illustrates that there may be a follow-up issue here to improve the level of translation here in the 15 foreign languages currently supported.  Bisq is a a truly global project!

The above also shows that redundant English strings in the foreign language files is **_very_** prevalent, and it certainly appears to have been a common practice, and it has been done according to the instructions. I'm just continuing the status quo by adding more. As a developer, I really feel uncomfortable using a tool like Google Translate to translate my new string in 15 languages when I have no ability to read and verify those translations. Like others before me, I'd prefer to just put the English in there as a placeholder until somebody, some day, with competency in that language undertakes the task that you mentioned of using a translation tool on the language file (and having the English strings in them already makes it much easier for such a person to use a translation tool like "transifex" that you mentioned to do it).

Reply:
Yes, I just told you that we use the translation tool Transifex and as far I am aware best practice is to only add the english string (default file) and let the translation be done by translators. The alternative path is to do it yourself with AI tools but of course as you said its not bulletproof (though also human translations had many mistakes). 
I am not familiar with the details of the workflow but I assume adding all languages in english adds the risk that the translation is missed. @ripcurlx might have more input here.

Reply:
As I saw now that we use other english entries, its probably best to stick with that...